src/Identity/Controllers/AccountController.cs

```diff
@@ -26,21 +28,61 @@ public class AccountController : Controller
         private readonly ILogger<AccountController> _logger;
         private readonly ISsoConfigRepository _ssoConfigRepository;
         private readonly IUserRepository _userRepository;
+        private readonly IOrganizationRepository _organizationRepository;
+        private readonly IHttpClientFactory _clientFactory;
 
         public AccountController(
             IClientStore clientStore,
             IIdentityServerInteractionService interaction,
             ILogger<AccountController> logger,
-            IOrganizationUserRepository organizationUserRepository,
             ISsoConfigRepository ssoConfigRepository,
             IUserRepository userRepository,
-            IUserService userService)
+            IOrganizationRepository organizationRepository,
+            IHttpClientFactory clientFactory)
         {
             _clientStore = clientStore;
             _interaction = interaction;
             _logger = logger;
             _ssoConfigRepository = ssoConfigRepository;
             _userRepository = userRepository;
+            _organizationRepository = organizationRepository;
+            _clientFactory = clientFactory;
+        }
+        
+        [HttpGet]
+        public async Task<IActionResult> PreValidate(string domainHint)
+        {
+            if (string.IsNullOrWhiteSpace(domainHint))
+            {
+                Response.StatusCode = 400;
+                return Json(new ErrorResponseModel("No domain hint was provided"));
+            }
+            try
+            {
+                // Calls Sso Pre-Validate, assumes baseUri set
+                var requestPath = $"/Account/PreValidate?domainHint={domainHint}";
+                var httpClient = _clientFactory.CreateClient("InternalSso");
+                using var responseMessage = await httpClient.GetAsync(requestPath);
```

Comment:
I see you added i18n to the sso service. Won't you need to forward the language header coming from the client?

Reply:
Done (now)